watchdog/src/fetch.rs

```diff
@@ -0,0 +1,184 @@
+use crate::bitcoin_block_apis::BitcoinBlockApi;
+
+/// The data fetched from the external bitcoin block APIs.
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct BlockInfo {
+    pub provider: BitcoinBlockApi,
```

Comment:
nit: Some doc comments on the fields would be nice.

Reply:
Done.